test/with_dummyserver/test_socketlevel.py

```diff
@@ -707,12 +707,13 @@ def socket_handler(listener):
         with HTTPConnectionPool(self.host, self.port) as pool:
             poolsize = pool.pool.qsize()
             timeout = Timeout(connect=LONG_TIMEOUT, read=SHORT_TIMEOUT)
-            response = pool.urlopen(
-                "GET", "/", retries=0, preload_content=False, timeout=timeout
-            )
             try:
-                with pytest.raises(ReadTimeoutError):
-                    response.read()
+                with pytest.raises((ReadTimeoutError, MaxRetryError)) as e:
```

Comment:
It's strange to me that we can raise a `MaxRetryError` when `retries=0`.

Reply:
`retries=False` is equivalent to `retries=Retry(total=False)`. My first commit was doing that, but I was getting connection timeouts, however I want to retry those at least once, which happens with `retries=0`.

Reply:
Shouldn't this only raise MaxRetryError then? Maybe I'm misunderstanding how our retry exhaustion behaves

Reply:
You're right, thanks! Fixed.